andela/ah-technocrats
# #161966609 users profile

### What does this pull request do?
 Add the users profile feature
### Description of the tasks to be completed?
- add functionality to enable users profile to be created upon successful registration.
 - add functionality to enable user to view profiles of other users and their own.
 - add functionality to enable users to edit only their profiles
 - add tests to test the above functionalities.
### How should this be manually tested?
- clone the repo `git clone https://github.com/andela/ah-technocrats.git`
- `cd ah-technocrats`
- checkout out to this branch `git checkout ft-user-profile-16196609 `
- make virtual environment `virtualenv venv`
- activate the virtual environment `source venv/bin/activate`
- install all the dependencies from the requirements file ` pip install -r requirements.txt`
- make migrations  `python manage.py makemigrations`
- apply the migrations `python manage.py migrate`
- run the server  `python manage.py runserver`
- go to Postman to confirm the functionalities.
-  the functionalities can also be tested automatically: run ` ./manage.py test authors/apps/profiles`
### Any background context you want to provide?
- Upon successful login of the user, the user profile should be automatically created. When a user is logged in, he/she should be allow to update the profile as one desires. 
- Authenticated users should be allow to see the profile of other users but should not be able to edit other user's profiles.
### What are the relevant Pivotal Tracker Stories
[#161966609](https://www.pivotaltracker.com/story/show/161966609)
# Checklist:
### GET USER PROFILE
<img width="1440" alt="screenshot 2018-12-04 at 21 00 59" src="https://user-images.githubusercontent.com/38049235/49508740-867d3180-f894-11e8-81fb-70dc91255ff2.png">

### UPDATE USER PROFILE
<img width="1440" alt="screenshot 2018-12-05 at 13 51 50" src="https://user-images.githubusercontent.com/38049235/49508890-f5f32100-f894-11e8-8ce9-95346f0cadf0.png">
